test/resource_test.rb

```diff
@@ -198,18 +358,25 @@ def test_destroy
     end
   end
 
+  def test_destroy_already_destroyed
```

Comment:
@sirupsen this is the regression test I added to ensure that there is no race to destroy the resource.